### DISCUSSION TITLE
Question after switching from the awesome thermostat

### DISCUSSION THREAD

**Discussion by Sourcekot** [jmcollin78/versatile_thermostat#294] [Q&A]
I'm currently in the process of switching from the awesome thermostat to the versatile thermostat and got a few questions:

1. The main reason for switching was the window delay but after reading the documentation it seems it is not working the way I expected it to work. What I want to achieve is that the heating doesn't start immediately after closing the window but after a delay of about 15 or 30 minutes as the room will reheat itself. Is that possible with VT`? I'm using mostly thermostats over climate with window detection in sensor mode.
2. As I'm switching from the awesome thermostat I'm wondering if there is something like the "hot_tolerance" or "cold_tolerance" or how is this handle in VT?
3. Is there a reason why there is no sleep mode like in the awesome thermostat? I thought this was supposed to be a rewrite of it so I'm surprised that some features seem to be missing.
4. Some of my thermostats don't allow an offset so I've handled them in my awesome thermostat via a switch which sets the target temperature to a few degrees higher than the real target temperature. Otherwise the thermostat itself would have detected a higher temperature than the target and switch itself off before the room really reaches the target.

I hope you can answer some or all of these questions for me.

UPDATE: I tried it now and its not working the way I expect it to work. I have a thermostat with a target temperature of 17.5°C and acurrent temperature of 19.3°C and still it is setting the associated climate entity to heating with a temperature of 18°C. As the target temperature is below the current temperature I expect it to simply turn off the climate entity or set its temperature to the frost preset as this is what awesome thermostat would do.

**Comment by jmcollin78** [comment]
Hello @Sourcekot ,

This is more question for the "discussions" rather than the Issues. I will convert it to discussion.

**Comment by jmcollin78** [comment]
Here are some quick answers:

1. No. There is short delay before considering the window as closed or open but no other delay on the delay to do what you want. I don't intend to implement more complexity over this case,
2. "hot_tolerance" or "cold_tolerance": i don't know exactly what it is refering about. If this is something like a threshold (above it is heating, under it is not heating), the answer is no. The regulation is smooth and power coefficient is recalculted at each cycle. There is no need of threshold and the current temperature is then much more stable. 
3. the sleep mode: I that a mode or a preset ? I think it is more a preset. The way preset are used in VTherm is much different. There is set of minimal preset (Frost, Eco, Comfort, Boost) which have different temperatures configuration considering the presence, the movement, the AC mode, ... So with 4 presets, you in fact have x4 preset (presence or not, AC or not) which are usable. And a static temperature configuration when power is overriden.
4. Some of my thermostats don't allow an offset: regulation algorithms are calculating the offset themself automatically and at each cycle. It is way better than having a static offset which cannot be relevant in the middle of the Winter and in the middle of the summer.

> I tried it now and its not working the way I expect it to work

> set its temperature to the frost preset as this is what awesome thermostat would do

I don't think awesome thermostat was doing that but that is not the point. Why should the Thermostat do that in your opinion ? I never see a thermostat switching alone the user preset because it reach the target temperature. The target temperature is set by the user and should not adapted by the thermostat. It is its goal for the regulation algorithm. The good behavior is, when the target is reach, how much power I send to maintain this target. This is the most efficient way to have a constant room temperature with the lower cost.
Suppose, I switch off, then the room temperature will fail and the user will be un happy.

When you change for another Thermostat, you have to adapt your mind to the new behavior. The question should not be: "is it working as I expect" but rather "Is the Thermostat doing a good regulation job" which is its goal.

If you need more tuning for adapting the regulation (there is some parameter to tune it), then I will need some curves as explained here: https://github.com/jmcollin78/versatile_thermostat/blob/main/README.md#even-better-with-plotly-to-tune-your-thermostat
Then you will have some regulation graphs like that: 
![Capture d’écran 2023-12-23 à 11 08 42](https://github.com/jmcollin78/versatile_thermostat/assets/1712374/96115000-0ae8-461a-abfc-8f571281df03)

Without this graph I cannot help for tuning the regulation.
